external_search.py

```diff
@@ -48,6 +56,35 @@ def reset(cls):
         """
         cls.__client = None
 
+    @classmethod
+    def search_index_update_operation(cls, _db):
+        """Determine the current name of the WorkCoverageRecord operation that
+        denotes "this work needs to be updated in the search index".
+
+        Because the name of the search index changes over time,
+        so does the name of the operation.
+        """
+        index_name = cls.works_index_name(_db) or ''
+        return (WorkCoverageRecord.UPDATE_SEARCH_INDEX_OPERATION
+                + '-' + index_name)
+
+    @classmethod
+    def search_integration(cls, _db):
+        """Look up the ExternalIntegration for ElasticSearch."""
+        return ExternalIntegration.lookup(
+            _db, ExternalIntegration.ELASTICSEARCH,
+            goal=ExternalIntegration.SEARCH_GOAL
+        )
+
+    @classmethod
+    def works_index_name(cls, _db):
+        """Look up the name of the search index."""
+        integration = cls.search_integration(_db)
+        if not integration:
+            return None
+        setting = integration.setting(cls.WORKS_INDEX_KEY)
+        return setting.value_or_default(cls.DEFAULT_WORKS_INDEX)
```

Comment:
I'm not sure about this. I think the index name in the database is usually the base index name, without a version number, and in some cases could be an alias instead. The code at the end of __init__ uses this setting to determine the actual versioned index name and the alias. We'd want to use the versioned index name in coverage records.